### PR TITLE
connectors-ci: unique secret name at session level

### PR DESCRIPTION
**Pull request by alafanechere**
## What
Relates to #27867 https://github.com/airbytehq/airbyte/issues/28617
According to the nightly tests results it appears that connectors spec validation tests often fails transiently, on different connectors.
My hypothesis is that the wrong config/secret is mounted to the CAT container [here](https://github.com/airbytehq/airbyte/blob/a29ca9b54fd3878eb79b18e7231a403007ad6d8f/tools/ci_connector_ops/ci_connector_ops/pipelines/actions/environments.py#L468)
After investigation it's probably related to a namespace issue [when we instantiate `Secret` object after GSM download:](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/actions/secrets.py#L59):
Dagger derives secret ID from their name and not their content, so far we've named secrets according to the config file name, which is not unique and often `config.json`. 
It means that when we call `context.dagger_client.set_secret(secret_filename, secret_plaintext)` in a concurrent context we can overwrite a connector secret with a wrong value as `dagger_client` is a pipeline wide object, not a context specific instance.

## How
Make secret name unique by concatenating connector name and file name:
```python
unique_secret_name = f"{context.connector.technical_name}_{secret_file}"
connector_secrets[secret_file] = context.dagger_client.set_secret(unique_secret_name, secret_plaintext)
```

## Tests
[Running a nightly build from this branch](https://github.com/airbytehq/airbyte/actions/runs/5654096657)

### CODE REVIEW

File: tools/ci_connector_ops/ci_connector_ops/pipelines/actions/secrets.py [hunk 1]
```diff
@@ -82,8 +84,9 @@ async def upload(context: ConnectorContext, gcp_gsm_env_variable_name: str = "GC
     ci_credentials = await environments.with_ci_credentials(context, gsm_secret)
 
     return await (
-        ci_credentials.with_directory(secrets_path, context.updated_secrets_dir)
-        .with_exec(["ci_credentials", context.connector.technical_name, "update-secrets"])
+        ci_credentials.with_directory(secrets_path, context.updated_secrets_dir).with_exec(
+            ["ci_credentials", context.connector.technical_name, "update-secrets"]
+        )
```
Comment (alafanechere) [inline tools/ci_connector_ops/ci_connector_ops/pipelines/actions/secrets.py:89]: formatting change

**Comment by erohmensing** [comment]
Is there still the chance that 2 concurrent runs for the same connector (one which modified the spec) could have this issue? 

E.g. the new 1s1t big query branch and the regular bigquery branch?

**Comment by alafanechere** [comment]
> Is there still the chance that 2 concurrent runs for the same connector (one which modified the spec) could have this issue?

@erohmensing By concurrent I mean concurrent within the same test pipeline execution. In our current paradigm a connector can't be tested twice is the same pipeline so I think we're safe.

**Comment by erohmensing** [comment]
Gotcha, so this is only a problem for changes with dependencies which kick off multiple test runs in the same session. Thanks for the clarification!

**Comment by alafanechere** [comment]
> Gotcha, so this is only a problem for changes with dependencies which kick off multiple test runs in the same session. Thanks for the clarification!

Or specifically nightly builds where we trigger tons of concurrent test execution in a single pipeline.

**Review by erohmensing — APPROVED** [review]
so long as the `connector_secrets` dict is per connector, looks good! otherwise i'd use the namespaced name as the key instead of just `secret_file`

**Comment by alafanechere** [comment]
Merging as the nightly build workflow, even if not finished, look healthy so far, and I'd like to see the result on the real tomorrow's nightly build.

**Pull request merged**